Req 05 show client info (with pet)

Billy-BonesUA opened the pull request:
Refactor: Improve output formatting for Pet and Dog classes

Enhance the information display by overriding the toString() method in both Pet and Dog classes. Additionally, provide implementations for the equals() and hashCode() methods for better object comparison and hashing.

Billy-BonesUA committed:
REQ-03: Email, First name, Last name validation create InputReader

Billy-BonesUA committed:
Restructured the project by adding packages, classes, and moving methods according to the video tutorial on "Inheritance and Encapsulation: Project Development."

Implemented the ability to add a pet and associate it with an owner. Modified the PetService, Client, and related classes to enable pet registration and association with a client.

Added functionality to check the immediate need to register a pet when registering a new client. Modified ClientService to register a new pet for a client, invoking the registerNewPet method from PetService. The registerPetForClient method now adds the pet to the PetRepository after successful registration.

Updated ApplicationRunner to invoke the new functionality during the client registration process.

Billy-BonesUA committed:
Correction class Valid

Billy-BonesUA committed:
Refactor: Improve output formatting for Pet and Dog classes
Enhance the information display by overriding the toString() method in both Pet and Dog classes. Additionally, provide implementations for the equals() and hashCode() methods for better object comparison and hashing.

InnaMel commented on src/main/java/com/magicvet/model/Dog.java:
```diff
@@ -0,0 +1,41 @@
+        return size;
+    }
+    @Override
+    public String toString() {
+        return String.format("Dog {type: %s, sex: %s, age: %s, name: %s, ownerName: %s, size: %s}",
+                getType(), getSex(), getAge(), getName(), getOwnerName(), size);
+    }
```
Are you sure that getType() != null in your case?

Billy-BonesUA committed:
Refactor: Improve output formatting for Pet and Dog classes
Enhance the information display by overriding the toString() method in both Pet and Dog classes. Additionally, provide implementations for the equals() and hashCode() methods for better object comparison and hashing.
